tests/src/texture/hipBindTex2DPitch.cpp

```diff
@@ -0,0 +1,101 @@
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+/*HIT_START
+ * BUILD: %t %s ../test_common.cpp
+ * TEST: %t
+ * HIT_END
+ */
+#include "test_common.h"
+#include<hip/hip_runtime.h>
+#include<hip/hip_runtime_api.h>
+#include<iostream>
+
+template<typename T> // pointer type
+void check(T input, T output, size_t height, size_t width)
+{
+    for(size_t i=0; i<height; i++ ){
+        for(size_t j=0; j<width; j++ ){
+            if( input[i*width + j] !=  output[ i*width + j ] ){
+                std::cout<<"input:"<<input[i*width + j]<<" output:"<<output[ i*width + j ];
+                failed("mistmatch at:%zu %zu",i,j);
+           }
+        }
+    }
+}
+
+#define SIZE_H 8
+#define SIZE_W 12
+#define TYPE_t float
+texture<TYPE_t, 2, hipReadModeElementType> tex;
+
+// texture object is a kernel argument
+__global__ void texture2dCopyKernel( TYPE_t* dst) {
+
+    int x = hipThreadIdx_x + hipBlockIdx_x * hipBlockDim_x;
+    int y = hipThreadIdx_y + hipBlockIdx_y * hipBlockDim_y;
+    if ( (x< SIZE_W) && (y< SIZE_H) ){
+        dst[SIZE_W*y+x] = tex2D(tex, x, y);
+    }
+}
+
+int main (void)
+{
+    TYPE_t* B;
+    TYPE_t* A;
+    TYPE_t* devPtrB;
+    TYPE_t* devPtrA;
+
+    B = new TYPE_t[SIZE_H*SIZE_W];
+    A = new TYPE_t[SIZE_H*SIZE_W];
+    for(size_t i=1; i <= (SIZE_H*SIZE_W); i++){
+        A[i-1] = i;
+    }
+
+    size_t devPitchA, tex_ofs;
+    HIPCHECK(hipMallocPitch((void**)&devPtrA, &devPitchA ,SIZE_W*sizeof(TYPE_t), SIZE_H)) ;
+    HIPCHECK(hipMemcpy2D(devPtrA, devPitchA, A, SIZE_W*sizeof(TYPE_t),
+            SIZE_W*sizeof(TYPE_t), SIZE_H, hipMemcpyHostToDevice));
+
+    tex.normalized = false;
+    #if defined(__HIP_PLATFORM_NVCC__)
+
+    cudaBindTexture2D(&tex_ofs, &tex, devPtrA, &tex.channelDesc,
+                                       SIZE_W, SIZE_H, devPitchA);
+    #else
+    HIPCHECK(hipBindTexture2D(&tex_ofs, &tex, devPtrA, &tex.channelDesc,
+                                       SIZE_W, SIZE_H, devPitchA));
+    #endif
+    HIPCHECK(hipMalloc((void**)&devPtrB, SIZE_W*sizeof(TYPE_t)*SIZE_H)) ;
+
+    hipLaunchKernelGGL(texture2dCopyKernel, dim3(4,4,1), dim3(32,32,1), 0, 0, devPtrB);
+    hipDeviceSynchronize();
+    HIPCHECK(hipMemcpy2D(B, SIZE_W*sizeof(TYPE_t), devPtrB, SIZE_W*sizeof(TYPE_t),
+            SIZE_W*sizeof(TYPE_t), SIZE_H, hipMemcpyDeviceToHost));
+
+    check(A, B, SIZE_H, SIZE_W);
+    delete []A;
+    delete []B;
+    hipFree(devPtrA);
+    hipFree(devPtrB);
+    passed();
```

Comment:
passed()/failed() should be decided based on check() hence it is good to have check() return true/false so that passed or failed can be decided.

Reply:
If execution reached here then test passed.